src/styles/text/canvas_text.js

```diff
@@ -343,3 +342,107 @@ CanvasText.font_size_re = /((?:[0-9]*\.)?[0-9]+)\s*(px|pt|em|%)?/;
 // Cache sizes of rendered text
 CanvasText.text_cache = {}; // by text style, then text string
 CanvasText.cache_stats = { hits: 0, misses: 0 };
+
+// Private class to arrange text labels into multiple lines based on
+// "text wrap" and "max line" values
+class MultiLine {
+    constructor (context, max_lines = Infinity, text_wrap = Infinity) {
+        this.width = 0;
+        this.height = 0;
+        this.lines = [];
+
+        this.ellipsis = '...';
```

Comment:
I wonder if we should use the Unicode single-character ellipsis? I suppose font support might be iffy, so maybe not.